lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -3707,3 +3733,75 @@ def get_test_data(delta=0.05):
     Y = Y * 10
     Z = Z * 500
     return X, Y, Z
+
+
+class _Quaternion:
+    """
+    Quaternions
+    consisting of scalar, along 1, and vector, with components along i, j, k
+    """
```

Comment:
I would add a bit more explanation that we are using unit quaternions, they are used for representing orientation, and a reference for interested readers to go learn more.

Reply:
Unlike numpy-quaternion which is more general, I think it makes sense to restrict the implementation here to unit quaternions which represent rotations, and disallow/not implement behavior which is only applicable in the general case.

Reply:
It adds complication, it is not needed for the purpose.
Also, is there even a good way to enforce the constraint (besides dividing by the (square root of the) norm and praying that the round-off errors won't spoil the effort)?

Reply:
I think the easy way to enforce the constraint would be to not allow initialization from the quaternion elements. Instead having the from_ methods being the only way to create them.

Reply:
I do think it’s safer to enforce the unit quaternion constraint, prevents issues later on. But since this is an internal API I don’t think it’s worth too much effort hammering out perfectly the scope of what we want this to be right now.